cli/bzk/service.go

```diff
@@ -74,115 +75,101 @@ func startService(cmd *cli.Cmd) {
+func stopService(cmd *cli.Cmd) {
+	cmd.Action = doStopService()
 }
 
-func doRestartService() {
-	client, err := docker.NewDocker("")
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	allContainers, err := client.Ps(&docker.PsOptions{
-		All: true,
-	})
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	config, err := getConfig()
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	if config.MongoURI == "" {
-		err = ensureContainerIsStarted(client, getMongoRunOptions(), allContainers)
+func doStopService() func() {
+	return func() {
+		client, err := docker.NewDocker("")
 		if err != nil {
 			log.Fatal(err)
 		}
-	}
 
-	tag, err := getTagFromCurrentImages(client, allContainers)
-
-	err = restartContainer(client, getServerRunOptions(config.Registry, config.Home, config.DockerSock, config.SCMKey, config.MongoURI, tag), allContainers)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	err = restartContainer(client, getWebRunOptions(config.Registry, tag), allContainers)
-	if err != nil {
-		log.Fatal(err)
-	}
-}
-
-func getTagFromCurrentImages(client *docker.Docker, allContainers []dockerclient.APIContainers) (string, error) {
-	container, err := getContainer(allContainers, bzkContainerServer)
-	if err != nil {
-		// Container not found, using latest by default
-		return "latest", nil
-	}
-	split := strings.Split(container.Image, ":")
-	return split[len(split)-1], nil
-}
-
-func upgradeService(cmd *cli.Cmd) {
-	cmd.Action = func() {
-		fmt.Printf("Pulling Bazooka images to check for new versions. This may take some time...\n")
-
-		client, err := docker.NewDocker("")
+		config, err := getConfig()
 		if err != nil {
 			log.Fatal(err)
 		}
 
-		bzkImages, err := client.Images(&docker.ImagesOptions{})
+		allContainers, err := client.Ps(&docker.PsOptions{
+			All: true,
+		})
 		if err != nil {
 			log.Fatal(err)
 		}
 
-		var hasError bool
-		for _, image := range bzkImages {
-			for _, tag := range image.RepoTags {
-				if strings.HasPrefix(tag, "bazooka/") {
-					err = client.Pull(&docker.PullOptions{Image: tag})
-					if err != nil {
-						fmt.Printf("Unable to pull image %s, reason is: %v\n", tag, err)
-						hasError = true
-					}
-					fmt.Printf("Newest image %s upgraded\n", tag)
-					break
-				}
+		if config.MongoURI == "" {
+			err := stopContainer(client, bzkContainerMongo, allContainers)
+			if err != nil {
+				log.Fatal(err)
 			}
 		}
-		if hasError {
-			log.Fatalf("Error while pulling some images, see above errors\n")
+
+		err = stopContainer(client, bzkContainerServer, allContainers)
+		if err != nil {
+			log.Fatal(err)
 		}
 
-		fmt.Printf("Bazooka Images have been upgraded, let's restart bazooka\n")
-		doRestartService()
+		err = stopContainer(client, bzkContainerWeb, allContainers)
+		if err != nil {
+			log.Fatal(err)
+		}
 	}
 }
 
-func stopService(cmd *cli.Cmd) {
-	cmd.Action = func() {
+func restartService(cmd *cli.Cmd) {
+	cmd.Spec = "[--recreate [-swd]]"
+
+	recreate := cmd.Bool(cli.BoolOpt{
+		Name:      "recreate",
```

Comment:
Maybe add a short name for this option to make it easier to chain ? `x` for example, to be able able to do `bzk service restart -xsw` for example